src/lib/one_or_two/intfs.ml

```diff
@@ -0,0 +1,11 @@
+type 'a t = [`One of 'a | `Two of 'a * 'a]
+
+module type Monadic = sig
```

Comment:
Could you instead use `Monad.S` from Base? And use `Monad.Make` there instead. Then we get Let_syntax for free too.

Reply:
huh? `One_or_two`s don't form a monad, this is the interface of monadic stuff you can do with `One_or_two`s. Like `Deferred.List.map` and friends.

Reply:
oh I see, the `t` is from the one or two. Okay